cylc/flow/graph_parser.py

```diff
@@ -525,10 +533,16 @@ def _proc_dep_pair(
         if right.count("(") != right.count(")"):
             raise GraphParseError(mismatch_msg.format(right))
 
-        # Ignore cycle point offsets on the right side.
-        # (Note we can't ban this; all nodes get process as left and right.)
+        # Raise error for cycle point offsets at the end of chains
         if '[' in right:
-            return
+            if left and (right in terminals):
+                # This right hand side is at the end of a chain:
+                raise GraphParseError(
+                    'ERROR, illegal cycle point offset on the right:'
```

Comment:
This kinda suggests that all offsets are illegal on the right, which isn't true. (See my comment on the change log line).

Also the `ERROR, ` prefix ends up being redundant (`GraphParseError: ERROR, illegal ...`).

Reply:
```suggestion
                    'Cycle point offsets only on the right of dependency pair:'
```

Reply:
Or how about
> Invalid cycle point offsets on right hand side of a dependency (must be on left hand side):

Reply:
Not true:

```
foo => bar[42]
bar[42] => baz
```
Is legal.

Reply:
Hence the "(must be on left hand side)"